### PR TITLE
[ADDED: C support + C pull consumer example

### PR DESCRIPTION
**Pull request by levb**
Preview: https://nats-by-example-mixf6pk98-connecteverything.vercel.app/examples/jetstream/pull-consumer/c

### CODE REVIEW

File: cmd/nbe/parse.go [hunk 1]
```diff
@@ -65,6 +65,7 @@ var (
 		DotNet:    "Main.cs",
 		DotNet2:   "Main.cs",
 		Elixir:    "main.exs",
+		C:         "main.c",
```
Comment (levb) [inline cmd/nbe/parse.go:68]: 1/5 leave it there, helps with debugging locally but not enough to make it an `nbe` flag

File: cmd/nbe/docker.go [hunk 2]
```diff
@@ -160,6 +160,7 @@ func (r *ImageBuilder) Run() (string, error) {
 	c := exec.Command(
 		"docker",
 		"build",
+		// "--progress=plain", // Use plain output for debuggig
```
Comment (levb) [inline cmd/nbe/docker.go:163]: 1/5 leave it there, helps with debugging locally but not enough to make it an `nbe` flag

File: examples/jetstream/pull-consumer/c/main.c [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+    // We set the batch size to 1000, but MaxBytes of 300 so we will only get 2
+    // messages at a time. Note that since we do not set NoWait, the call will
+    // block until the batch is filled or the timeout expires, unlike
+    // `natsSubscription_Fetch`.
+    for (ibatch = 0, c = 0; (s == NATS_OK) && (c < NUM_MESSAGES); ibatch++)
+    {
+        int64_t start = nats_Now();
+        jsFetchRequest fr = {
+            .Batch = 1000,
+            // .NoWait = true,
```
Comment (bruth) [inline examples/jetstream/pull-consumer/c/main.c:226]: Suggest removing this since it shows up as text when the example renders.

Comment (levb) [reply examples/jetstream/pull-consumer/c/main.c]: Updated to a /**/ with a note.

**Pull request merged**